servicetalk-utils-internal/src/main/java/io/servicetalk/utils/internal/PlatformDependent0.java

```diff
@@ -100,16 +113,56 @@ final class PlatformDependent0 {
             // is an instanceof Unsafe and reversing the if and else blocks; this is because an
             // instanceof check against Unsafe will trigger a class load and we might not have
             // the runtime permission accessClassInPackage.sun.misc
-            if (maybeUnsafe instanceof Exception) {
+            if (maybeUnsafe instanceof Throwable) {
                 unsafe = null;
-                LOGGER.debug("sun.misc.Unsafe.theUnsafe: unavailable", (Exception) maybeUnsafe);
+                LOGGER.debug("sun.misc.Unsafe.theUnsafe: unavailable", (Throwable) maybeUnsafe);
             } else {
-                unsafe = (Unsafe) maybeUnsafe;
-                LOGGER.debug("sun.misc.Unsafe.theUnsafe: available");
+                unsafe = maybeUnsafe;
+                LOGGER.debug("sun.misc.Unsafe.theUnsafe: {}", null == unsafe ? "unavailable" : "available");
             }
         }
         UNSAFE = unsafe;
 
+        if (null == unsafe) {
+            THROW_EXCEPTION = null;
+            ALLOCATE_MEMORY = null;
+            FREE_MEMORY = null;
+        } else {
+            Consumer<Throwable> throwConsumer = null;
+            MethodHandle allocateMH = null;
+            MethodHandle freeMH = null;
+            MethodHandles.Lookup lookup = MethodHandles.lookup();
+            MethodHandle throwExceptionMH = null;
+            try {
+                throwExceptionMH = lookup.findVirtual(unsafe.getClass(), "throwException",
+                        MethodType.methodType(void.class, Throwable.class));
+                CallSite throwExceptionCallSite = LambdaMetafactory.metafactory(
```

Comment:
What is the benefit of using a lambda here instead of a direct use of a `MethodHandle` like in all other cases?

Reply:
My thought was to use it in `PlatformDependent` by having a static field that selected which implementation to use based on the availability of Unsafe. The sneakyThrows would be referenced using a method reference.

I didn't end up making that change though so this could be used as a reference handle. The JIT compiler is also reputedly better about optimizing/inlining `CallSite` functional interfaces than `MethodHandles`

Reply:
Ok, thanks for explaining. Don't mind either way, was just curious.

Reply:
I did end up using this technique in the updated PR.